### PR TITLE
+ pkgdown + testthat + MIT + reduced to single CRAN version example

### PR DESCRIPTION
**Pull request by kkmann**
Hey, 
this is a major overhaul of the basic infrastructure. I added testthat infrastructure, automatic code coverage reports (needs set up by repository owner), automatic pkgdown deployment (needs setup, see here https://pkgdown.r-lib.org/reference/deploy_site_github.html) and reduced the example to a bare minimum to get rid of the bioconductor dependencies.
This still needs some tinkering before the CI/CD + codecov + pkgdown works smoothly, @elong0527 can you set that up at some point?

### CODE REVIEW

**Comment by kkmann** [comment]
sorry for messing up the master branch before ;) you might want to set up a branch protection rule there ;)

File: tests/testthat/test_cran_metrics.R [hunk 1]
```diff
@@ -0,0 +1,21 @@
+context("Test CRAN related metrics")
+
+test_that("CRAN current version returns correct format", {
+
+    expect_equal(
+      grep(
+        "[0-9]+.[0-9]+.[0-9]+(.[0-9]+)*",
```
Comment (dgkf) [inline tests/testthat/test_cran_metrics.R]: Be careful with the `.` usage here - `.` is a reserved character in `regex` and will capture any characters such that "1a2b3cd" would be a valid version number. Not a huge deal, and the standard version format will still get picked up. 

Consider using `"\\d+(?:\\.\\d+){2,}"`

File: R/memoise_cran_db.R [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# taken from https://github.com/ropenscilabs/packagemetrics/blob/master/R/get_cran.R
+memoise_cran_db <- memoise::memoise({
```
Comment (dgkf) [inline R/memoise_cran_db.R]: Super cool. I'll be using this! I always write an annoying amount of boilerplate to handle this type of thing.

File: tests/testthat/test_cran_metrics.R (hunk 1 above)
Comment (kkmann) [reply tests/testthat/test_cran_metrics.R]: oh dear, regular expressions certainly are not my strong suit ;) thx for pointing that out!

File: R/memoise_cran_db.R (hunk 2 above)
Comment (kkmann) [reply R/memoise_cran_db.R]: functional programming roxxx ;)

**Comment by kkmann** [comment]
@elong0527 since this is almost a complete overhaul, I'll wait for your opinion as well. can you set up all the ci/cd stuff after merging?

**Comment by elong0527** [comment]
@kkmann , I am waiting for the permission approve from @andyofsmeg to deploy the pkgdown. Will complete the set up after approval.

**Pull request merged**